arkouda/dataframe.py

```diff
@@ -2102,6 +2102,101 @@ def unregister_dataframe_by_name(user_defined_name: str) -> None:
         unregister_pdarray_by_name(f"df_index_{user_defined_name}_key")
         Strings.unregister_strings_by_name(f"df_columns_{user_defined_name}")
 
+    @staticmethod
+    def _parse_col_name(entryName, dfName):
+        """
+        Helper method used by from_return_msg to parse the registered name of the data component
+        and pull out the column type and column name
+
+        Parameters
+        ----------
+        entryName : string
+            The full registered name of the data component
+
+        dfName : string
+            The name of the DataFrame
+
+        Returns
+        -------
+        Tuple (columnName, columnType)
+        """
+        regName = entryName.split(" ")[1]
+        colParts = regName.split("_")
+        colType = colParts[2]
+
+        # Case of '_' in the column or dataframe name
+        if len(colParts) > 5:
+            nameInd = regName.rindex(dfName) - 1
+            startInd = len(colType) + 9
```

Comment:
what purpose does the `+ 9` serve?

Reply:
It pads the start index so you're only looking at the column name instead of having to do any additional parsing of the string. An example formatting and break down of this for this case would be `df_data_column_name_dfName`. This takes a slice after the `df_data_` (which is always there, and can't vary in length) until the `_dfName` which results in `column_name` regardless of special characters in either the column or dataframe name